grr/server/grr_response_server/gui/ui/components/timestamp/timestamp_test.ts

```diff
@@ -0,0 +1,106 @@
+import {async, TestBed} from '@angular/core/testing';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {initTestEnvironment} from '@app/testing';
+
+import {TimestampModule} from './module';
+import {Timestamp} from './timestamp';
+import {DatePipe} from '@angular/common';
+
+initTestEnvironment();
+
+describe('Timestamp Component', () => {
+  const ABSOLUTE_FORMAT: string = "MMM d \''yy 'at' HH:mm";
+
+  beforeEach(async(() => {
+    TestBed
+      .configureTestingModule({
+        imports: [
+          TimestampModule,
+          NoopAnimationsModule,  // This makes test faster and more stable.
+        ],
+      })
+      .compileComponents();
+  }));
+
+  it('is created successfully', () => {
+    const fixture = TestBed.createComponent(Timestamp);
+    const componentInstance = fixture.componentInstance;
+    expect(componentInstance).toBeTruthy();
+  });
+
+  it('shows "Just now" for dates less than 30s ago', () => {
```

Comment:
In tests, generally test for edge cases. For your time handling logic, please test the biggest value below the boundary (29.999s) and exactly the boundary (30s). For added fun, test some very small numbers (e.g. 0), very big numbers (e.g. 1000 years ago), and fractional numbers (e.g. 44.444s).